tasks/workflow/step.go

```diff
@@ -371,22 +372,48 @@ func (s *step) runActivity(wfCtx context.Context, cfg config.Configuration, depl
 		}
 	case builder.ActivityTypeInline:
 		// Register inline workflow associated to the original task
-		return s.registerInlineWorkflow(wfCtx, activity.Value())
+		return s.registerInlineWorkflow(wfCtx, deploymentID, activity.Value())
 	}
 	return nil
 }
 
-func (s *step) registerInlineWorkflow(ctx context.Context, workflowName string) error {
+func (s *step) registerInlineWorkflow(ctx context.Context, deploymentID, workflowName string) error {
 	events.WithContextOptionalFields(ctx).NewLogEntry(events.LogLevelINFO, s.t.targetID).RegisterAsString(fmt.Sprintf("Register workflow %q from taskID:%q, deploymentID:%q", workflowName, s.t.taskID, s.t.targetID))
-	wfOps, err := builder.BuildInitExecutionOperations(ctx, s.t.targetID, s.t.taskID, workflowName, true)
+
+	// Preparing a new task with its own data referencing the parent workflow step
+	parentData, err := tasks.GetAllTaskData(s.t.taskID)
 	if err != nil {
 		return err
 	}
-	err = tasks.StoreOperations(s.t.taskID, wfOps)
+	data := make(map[string]string)
+	for k, v := range parentData {
+		data[k] = v
+	}
+	data[taskDataParentWorkflowName] = s.WorkflowName
+	data[taskDataParentStepName] = s.Name
+	data[taskDataParentTaskID] = s.t.taskID
+	data[taskDataDeploymentID] = deploymentID
+	data[taskDataWorkflowName] = workflowName
+
+	taskID, err := collector.NewCollector(s.cc).RegisterTaskWithData(deploymentID, tasks.TaskTypeCustomWorkflow, data)
 	if err != nil {
-		err = errors.Wrapf(err, "Failed to register workflow init operations with workflow:%q, targetID:%q, taskID:%q", workflowName, s.t.targetID, s.t.taskID)
+		err = errors.Wrapf(err, "Failed to register inline workflow %s in parent workflow %s step %s, targetID %s, taskID %s",
+			workflowName, s.WorkflowName, s.Name, s.t.targetID, s.t.taskID)
+	} else {
+		log.Debugf("Registered task %s for inline workflow %s in parent workflow %s step %s",
+			taskID, workflowName, s.WorkflowName, s.Name)
+		err = s.setStatus(tasks.TaskStepStatusRUNNING)
 	}
 
+	if err != nil {
+		err = s.setStatus(tasks.TaskStepStatusERROR)
+
+	}
+	// Marking this step as asynchronous as it should not be considered as
+	// done by the caller
+	s.Async = true
```

Comment:
Does async step needs any monitoring action ?

Reply:
No, the inline workflow when it ends will take care of updating this parent step in parent workflow

Reply:
OK